src/colmap/estimators/generalized_absolute_pose.cc

```diff
@@ -50,33 +50,30 @@ void GP3PEstimator::Estimate(const std::vector<X_t>& points2D,
   THROW_CHECK_NOTNULL(rigs_from_world);
 
   std::vector<Eigen::Vector3d> rays_in_rig(3);
-  std::vector<Eigen::Vector3d> origins_in_world(3);
+  std::vector<Eigen::Vector3d> origins_in_rig(3);
   for (int i = 0; i < 3; ++i) {
     const Rigid3d rig_from_cam = Inverse(points2D[i].cam_from_rig);
     rays_in_rig[i] =
         (rig_from_cam.rotation * points2D[i].ray_in_cam).normalized();
-    origins_in_world[i] = rig_from_cam.translation;
+    origins_in_rig[i] = rig_from_cam.translation;
   }
 
   std::vector<poselib::CameraPose> poses;
-  int num_poses = 0;
-  if (points2D[0].cam_from_rig.translation.isApprox(
-          points2D[1].cam_from_rig.translation, 1e-9) &&
-      points2D[0].cam_from_rig.translation.isApprox(
-          points2D[2].cam_from_rig.translation, 1e-9)) {
+  if (origins_in_rig[0].isApprox(origins_in_rig[1], 1e-6) &&
```

Comment:
This was a bug. It would only work when the camera origins coincide with the rig origin.